SU2_CFD/include/variables/CDiscAdjVariable.hpp

```diff
@@ -45,192 +45,186 @@
+    return Geometry_CrossTerm_Derivative_Flow(iPoint,iVar);
+  }
 
   /*!
    * \brief Set the value of the mesh cross term derivative from the flow solution (adjoint).
    * \param[in] der - cross term derivative.
    */
-  inline void SetGeometry_CrossTerm_Derivative_Flow(unsigned short iDim, su2double der) {Geometry_CrossTerm_Derivative_Flow[iDim] = der;}
+  inline void SetGeometry_CrossTerm_Derivative_Flow(Idx_t iPoint, Idx_t iDim, su2double der) override {
+    Geometry_CrossTerm_Derivative_Flow(iPoint,iDim) = der;
+  }
 
   /*!
    * \brief Set the value of the mesh solution (adjoint).
-   * \param[in] val_solution - Solution of the problem (acceleration).
    */
-  inline void Set_OldSolution_Geometry(void) {
-    for (unsigned short iDim = 0; iDim < nDim; iDim++)
-      Solution_Geometry_Old[iDim] = Solution_Geometry[iDim];
-  }
+  void Set_OldSolution_Geometry() override;
 
   /*!
    * \brief Get the value of the old geometry solution (adjoint).
    * \param[out] val_solution - old adjoint solution for coordinate iDim
    */
-  inline su2double Get_OldSolution_Geometry(unsigned short iDim) {return Solution_Geometry_Old[iDim];}
+  inline su2double Get_OldSolution_Geometry(Idx_t iPoint, Idx_t iDim) const override {
+    return Solution_Geometry_Old(iPoint,iDim);
+  }
 
   /*!
    * \brief Set the value of the adjoint solution in the current BGS subiteration.
    */
-  inline void Set_BGSSolution(unsigned short iDim, su2double val_solution) {Solution_BGS[iDim] = val_solution;}
+  inline void Set_BGSSolution(Idx_t iPoint, Idx_t iDim, su2double val_solution) override {
+    Solution_BGS(iPoint,iDim) = val_solution;
+  }
 
   /*!
    * \brief Get the value of the adjoint solution in the previous BGS subiteration.
    * \param[out] val_solution - adjoint solution in the previous BGS subiteration.
    */
-  inline su2double Get_BGSSolution(unsigned short iDim) {return Solution_BGS[iDim];}
-
-  /*!
-   * \brief Set the value of the adjoint geometry solution in the previous BGS subiteration.
-   */
-  inline void Set_BGSSolution_Geometry(void) {
-    for (unsigned short iDim = 0; iDim < nDim; iDim++)
-      Solution_Geometry_BGS_k[iDim] = Solution_Geometry[iDim];
-  }
-
-  /*!
-   * \brief Get the value of the adjoint geometry solution in the previous BGS subiteration.
-   * \param[out] val_solution - geometrical adjoint solution in the previous BGS subiteration.
-   */
-  inline su2double Get_BGSSolution_Geometry(unsigned short iDim) {return Solution_Geometry_BGS_k[iDim];}
+  inline su2double Get_BGSSolution(Idx_t iPoint, Idx_t iDim) const override { return Solution_BGS(iPoint,iDim);}
 
   /*!
    * \brief Set the contribution of crossed terms into the derivative.
    */
-  inline void SetCross_Term_Derivative(unsigned short iVar, su2double der) {Cross_Term_Derivative[iVar] = der; }
+  inline void SetCross_Term_Derivative(Idx_t iPoint, Idx_t iVar, su2double der) override {
```

Comment:
I am not sure about other systems, but on mac/llvm, the override keyword is resulting in many warnings due to inconsistent/missing override statements on some functions... do you see these when you build this up?

Reply:
Nothing on gcc... I'll try to replicate, if I can I'll fix it.

Reply:
With clang 3.8 I get no warning from CVariable files with -Wall -Wextra -Wno-unused-parameter.
I do get override warnings from other files due to the absence of the keyword (-Winconsistent-missing-override) clang is probably turning on this best-practices-based warning due to the c++11 flag.

Reply:
Guess we have some cleanup to do then. 

Also, still plan to search and replace 'Idx_t'?

Reply:
Yup very last thing before merging as it makes it easier to spot new functions from other PR's
How do you propose handling #777 and #790?

Reply:
#790 is more or less complete in terms of content. Still need to update a couple of items in the output. #777 will need a bit of work now that #724 is merged and after this PR, but that is ok. In fact, I am fine if we do not rush just to get that into the release.

So, I would say we go ahead and merge this and #774 very soon. Then I can update #790 (only the new memory for the reconstruction gradient is affected), and we can get that in afterwards. I'll keep working on #777, but let's take it out of the critical path for the release (it is all new functionality anyway).

Reply:
Ok, sounds good, regardless of merging order I can help. I still want to try #790 on the benchmark case I used for this PR, from your results so far we'll gain more from that than from the memory layout.